kernel/virtio/virtio_blk.c

```diff
@@ -183,47 +183,42 @@ void virtio_config_block(struct pci_config_info *pci)
     outb(pci->base + VIRTIO_PCI_STATUS, VIRTIO_PCI_STATUS_DRIVER_OK);
 }
 
-int solo5_blk_write_sync(uint64_t sector, uint8_t *data, int n)
+solo5_result_t solo5_block_write(solo5_lba_t lba, const uint8_t *buf,
+        size_t size)
 {
     assert(blk_configured);
     /*
      * XXX: virtio_blk_op_sync does not support >1 sectors per operation.
      */
-    if (n > VIRTIO_BLK_SECTOR_SIZE)
-        return -1;
+    if (size > VIRTIO_BLK_SECTOR_SIZE)
```

Comment:
virtio (qemu size) fails IOs smaller than SECTOR_SIZE. So for those, we end up returning an `SOLO5_R_EUNSPEC` (below). It might be simpler to check for IO's being of `SECTOR_SIZE` now, and return an `SOLO5_R_EINVAL` instead.